src/Rewrite.hs

```diff
@@ -81,3 +83,45 @@ quotedUrls log (Args _ attrPth drvFile _) = do
     else do
       lift $ log "[quotedUrls] nothing found to replace"
       return Nothing
+
+--------------------------------------------------------------------------------
+-- One time migration for #79975
+rustCargoFetcher :: MonadIO m => (Text -> m ()) -> Args -> ExceptT Text m (Maybe Text)
+rustCargoFetcher log (Args _ _ drvFile drvContents) = do
+  lift $ log "[rustCargoFetcher] started"
+  if not (T.isInfixOf "Delete this on next update; see #79975" drvContents)
+    then do
+      lift $ log "[rustCargoFetcher] nothing to do"
+      return Nothing
+    else do
+      _ <- readProcess $ proc "sed" ["-i", "/.*Delete this on next update; see #79975 for details/,/^$/d", drvFile]
+      lift . log $ "[rustCargoFetcher] Ran sed on " <> T.pack drvFile
+      -- N.B. The actual hash fix will come from the next rewriter!
+      return $ Just "Updated cargo fetcher for #79975"
+
+--------------------------------------------------------------------------------
+-- Rewrite Rust on rustPlatform.buildRustPackage
+-- This is basically rewriteVersion above, but we do a second pass for the cargoSha256 vendor hash.
+rustCrateVersion :: MonadIO m => (Text -> m ()) -> Args -> ExceptT Text m (Maybe Text)
+rustCrateVersion log (Args env attrPth drvFile drvContents) = do
+  lift $ log "Processing: rewriteRustCrateVersion"
+  if not (T.isInfixOf "cargoSha256" drvContents)
+    then do
+      lift $ log "No cargoSha256 found"
+      return Nothing
+    else do
+      oldHash <- Nix.getOldHash attrPth
+      -- This starts the same way rewriteVersion does, minus the assert
+      _ <- lift $ File.replace (Utils.oldVersion env) (Utils.newVersion env) drvFile
+      _ <- lift $ File.replace oldHash Nix.sha256Zero drvFile
+      newHash <- Nix.getHashFromBuild attrPth
+      _ <- lift $ File.replace Nix.sha256Zero newHash drvFile
+      -- But then from there we need to do this a second time!
+      oldCargoSha256 <- Nix.getDrvAttr "cargoSha256" attrPth
+      _ <- lift $ File.replace oldCargoSha256 Nix.sha256Zero drvFile
+      newCargoSha256 <- Nix.getHashFromBuild attrPth
+      lift . log $ "Replacing cargoSha256 with " <> newCargoSha256
+      _ <- lift $ File.replace Nix.sha256Zero newCargoSha256 drvFile
+      Nix.build attrPth
```

Comment:
I think we should probably add in the checks that both hashes actually changed.

Reply:
Why is Nix.build being called here?

Reply:
Definitely, added the checks!  Also fixed the log msgs.

I was calling `Nix.build` to ensure our newly modified package actually builds and passes its tests.  This is technically done at the end prior to calling publish, so we could omit it; but if we don't compile here it's nice to throw a failure close to where it occurred.